### ISSUE TITLE
Revamp CI based on split config

### ISSUE THREAD

**Issue by arnetheduck** [status-im/nim-beacon-chain#2726]
With [split config](https://github.com/status-im/nimbus-eth2/pull/2710) implemented, there are predominantly two kinds of binaries that cover 99% of all use cases - mainnet and minimal.

As such, we can revamp most of the CI pipeline to create only these two binary kinds, then run all of rest, local sims, test etc using the same binaries - this will also enable running a lot more steps in parallel. To implement this:

* `minimal` binaries should be built a `minimal` suffix - mainnet binaries remain "suffixless"
* scripts like `launch_local_testnet.sh` etc should never perform builds - instead, they should run things with whatever binaries exist already
* `make` should be used to get "build+run" behavior, inverting the current behavior where the script runs make
* "special" build flags like "local_testnet" can be removed - they were relevant at some stage during development but now represent debt mainly, and make life harder by producing slightly unique binaries for no particular reason

**Comment by arnetheduck** [comment]
No longer relevant
